src/components/Subthoughts.js

```diff
@@ -226,6 +228,24 @@ export const Subthoughts = connect(({ cursorBeforeEdit, cursor, contextViews, th
           : showContexts ? getContextsSortedAndRanked(/* subthought() || */headValue(thoughtsRanked))
             : getThoughtsRanked(contextBinding || thoughtsRanked)
 
+      // Ensure that editable newThought is visible.
+      const editIndex = (cursor && children && show) ? children.findIndex(child => {
+        const childPath = getChildPath(child, thoughtsRanked, showContexts)
+        const isEditingPath = subsetThoughts(cursor, childPath)
+        return equalPath(cursor, childPath) || isEditingPath
+      }) : 0
```

Comment:
What about storing the index in `cursor`? This would eliminate the need to find the matching child path to determine the page cutoff.

Reply:
didn't get the question.

Reply:
`cursor` currently contains `{ rank, value }`. If we changed the `cursor` type to `{ rank, value, index }`, then it would save us a lot of time here.

Reply:
actually saving index in the cursor will not actually save us anything here as here we have to check not only where the cursor but also are the paths of cursor and child are equal or not. 

And basically I think it will add unnecessary complexity to the code and the output is not worth it.

Reply:
I'm still concerned about the performance of this approach. We can leave it for now so as not to complicate this PR, but it may need to be refactored if it proves to add unnecessary cycles.